frame/contracts/src/lib.rs

```diff
@@ -1288,11 +1338,12 @@ impl<T: Config> Pallet<T> {
 sp_api::decl_runtime_apis! {
 	/// The API used to dry-run contract interactions.
 	#[api_version(2)]
-	pub trait ContractsApi<AccountId, Balance, BlockNumber, Hash> where
+	pub trait ContractsApi<AccountId, Balance, BlockNumber, Hash, EventRecord> where
```

Comment:
Doesn't this change break the current API require a version bump?

Reply:
Yes. While adding trailing data might work we should bump the version. We should also support the old version with the unchanged struct for some time.

Reply:
It seems like `decl_runtime_apis` does not support multiple declarations of the same trait (it only support multiple declarations of functions). Would you suggest to create a new trait with a different name for the support of the new `EventRecord`? if so, should it be something like `ContractsApiV3` and add `#[api_version(3)]`?

Reply:
After talking about it in DM we have decided to create a new trait `ContractsApiV3`